js/views/ListingCard.js

```diff
@@ -108,55 +107,63 @@ export default class extends baseVw {
   onClickEdit(e) {
     app.loadingModal.open();
 
-    const fullListingFetch = this.fullListing.fetch()
-      .done(() => {
-        if (fullListingFetch.statusText === 'abort' || this.isRemoved()) return;
+    this.fetchFullListing()
+      .done(xhr => {
+        if (xhr.statusText === 'abort' || this.isRemoved()) return;
 
-        this.editModal = launchEditListingModal({
+        launchEditListingModal({
           model: this.fullListing,
         });
       })
       .always(() => {
         if (this.isRemoved()) return;
         app.loadingModal.close();
-      })
-      .fail(xhr => {
-        let failReason = xhr.responseJSON && xhr.responseJSON.reason || '';
-
-        if (xhr.status === 404) {
-          failReason = app.polyglot.t('listingCard.editFetchErrorDialog.bodyNotFound');
-        }
-
-        openSimpleMessage(
-          app.polyglot.t('listingCard.editFetchErrorDialog.title'),
-          failReason
-        );
       });
 
     e.stopPropagation();
   }
 
-  onClickDelete() {
+  onClickDelete(e) {
     this.getCachedEl('.js-deleteConfirmedBox').removeClass('hide');
     this.deleteConfirmOn = true;
-    // don't bubble to the document click handler
-    return false;
+    e.stopPropagation();
   }
 
-  onClickConfirmedDelete() {
-    if (this.destroyRequest && this.destroyRequest.state === 'pending') return false;
+  onClickClone(e) {
+    app.loadingModal.open();
+
+    this.fetchFullListing()
+      .done(xhr => {
+        if (xhr.statusText === 'abort' || this.isRemoved()) return;
```

Comment:
Same comment here, if the .always callback was added first, couldn't it handle all the checking for abort and isRemoved?